### ISSUE TITLE
retain

### ISSUE THREAD

**Issue by bowenwang1996** [jeromefroe/lru-rs#31]
It would be nice to add a retain method, similar to what `std::collection::HashMap` and `std::vec::Vec` have.

**Comment by jeromefroe** [comment]
I agree, that would be helpful! I think I should have time in the next couple of weeks to take a crack at it. PR's are certainly welcome though if you'd like to take a look at it before then :)
